relay-server/src/services/processor.rs

```diff
@@ -1274,23 +1274,41 @@ impl EnvelopeProcessorService {
         event::finalize(state, &self.inner.config)?;
         self.light_normalize_event(state)?;
         dynamic_sampling::normalize(state);
-        event::filter(state)?;
-        dynamic_sampling::run(state, &self.inner.config);
+        event::filter(state, &self.inner.global_config.current())?;
+        // Don't extract metrics if relay can't apply generic inbound filters.
+        // An inbound filter applied in another up-to-date relay in chain may
+        // need to drop the event, and there should not be metrics from dropped
+        // events.
+        // In processing relays, always extract metrics to avoid losing them.
+        let supported_generic_filters = self.inner.global_config.current().filters.is_ok()
```

Comment:
I really think `event::filter` should return this, this is filter related logic and just hard to read here.

Reply:
Makes sense. I'll address this in a follow-up so that this PR can already land.